v3/src/components/graph/components/graph.tsx

```diff
@@ -233,13 +233,24 @@ export const Graph = observer(function Graph({graphController, graphRef, pixiPoi
 
   // respond to assignment of new attribute ID
   useEffect(function handleNewAttributeID() {
-    const disposer = graphModel && onAnyAction(graphModel, action => {
-      if (isSetAttributeIDAction(action)) {
-        startAnimation()
-        graphController?.handleAttributeAssignment()
-      }
-    })
-    return () => disposer?.()
+    const react = () => {
+      startAnimation()
+      graphController?.handleAttributeAssignment()
+    }
+    let disposer: Maybe<IDisposer>
+    let disposer2: Maybe<IDisposer>
+    if (graphModel) {
+      disposer = onAnyAction(graphModel, action => {
+        if (isSetAttributeIDAction(action)) react()
+      })
+      disposer2 = onAnyAction(graphModel.dataConfiguration, action => {
+        if (["setAttribute", "replaceYAttribute", "removeYAttributeAtIndex"].includes(action.name)) react()
+      })
```

Comment:
`onAnyAction` notifications "bubble" up through the MST tree, so it should be possible to listen to `"setAttribute", "replaceYAttribute", "removeYAttributeAtIndex"` actions on the graph model as well, i.e. you _should_ be able to accomplish all of this with a single `onAnyAction` on the `graphModel`.